cmake/StdlibFamilies.cmake

```diff
@@ -0,0 +1,178 @@
+#
+#  Copyright (c) 2025, Intel Corporation
+#
+#  SPDX-License-Identifier: BSD-3-Clause
+
+# This file defines stdlib families used by both CMake (for stdlib compilation)
+# and C++ (for runtime lookup). Each family maps multiple targets to a single
+# representative (base) target whose stdlib they share.
+#
+# Two types of base targets:
+#   1. generic-* targets: For families where targets produce stdlib identical to generic
+#   2. Real ISA targets: For inheritance families where child ISAs produce stdlib
+#      identical to parent ISA (but different from generic, usually due to alignment)
+#
+
+# Define width families in a unified declarative format
+# Format: "family_name:base_target:member1,member2,member3,..."
+# where base_target is either:
+#   - generic-* for families that match generic implementation
+#   - real ISA target for inheritance families (e.g., avx1-i32x8 for avx2/avx2vnni)
+#
+# This function creates variables for each width family:
+#   STDLIB_FAMILY_<family_name>: List of member targets
+#   STDLIB_FAMILY_BASE_<family_name>: Base target
+#   STDLIB_FAMILIES: List of all family names
+#   STDLIB_FAMILY_ALL_MEMBERS: Flat list of all targets in any family
+function(define_stdlib_families)
+    set(all_families "")
+    set(all_members "")
+
+    # Single declarative list of all families
+    # Note: nozmm targets are excluded - they produce different stdlib that avoids ZMM instructions
+    set(FAMILY_DEFINITIONS
+        # AVX512 families - use generic targets
+        "i1x4:generic-i1x4:avx512skx-x4,avx512icl-x4,avx512spr-x4"
+        "i1x8:generic-i1x8:avx512skx-x8,avx512icl-x8,avx512spr-x8"
+        "i1x16:generic-i1x16:avx512skx-x16,avx512icl-x16,avx512spr-x16"
+        "i1x32:generic-i1x32:avx512skx-x32,avx512icl-x32,avx512spr-x32"
+        "i1x64:generic-i1x64:avx512skx-x64,avx512icl-x64,avx512spr-x64"
+
+        # SSE/AVX families that match generic implementation
+        "i32x4:generic-i32x4:sse2-i32x4,sse4-i32x4"
+        "i32x8:generic-i32x8:sse2-i32x8,sse4-i32x8"
+        "i32x16:generic-i32x16:avx1-i32x16,avx2-i32x16,avx2vnni-i32x16"
+        "i64x4:generic-i64x4:avx1-i64x4,avx2-i64x4"
+        "i16x8:generic-i16x8:sse4-i16x8"
+        "i16x16:generic-i16x16:avx2-i16x16"
+        "i8x16:generic-i8x16:sse4-i8x16"
+        "i8x32:generic-i8x32:avx2-i8x32"
+
+        # Inheritance families - use parent ISA target (NOT generic!)
+        # These produce stdlib different from generic but identical to parent ISA
+        "avx_i32x8:avx1-i32x8:avx2-i32x8,avx2vnni-i32x8"
+        "avx_i32x4:avx2-i32x4:avx2vnni-i32x4"
+    )
+
+    # Add AVX10 targets to families if LLVM supports them (>= 20.1.2)
+    if (${LLVM_VERSION_NUMBER} VERSION_GREATER_EQUAL "20.1.2")
+        # Append AVX10 targets to existing AVX512 families
+        list(APPEND FAMILY_DEFINITIONS
+            "i1x4_avx10:generic-i1x4:avx10_2-x4"
```

Comment:
Should we just append it to the previously declared `"i1x4:generic-i1x4:avx512skx-x4,avx512icl-x4,avx512spr-x4"`? It seems to be from the same family for me.

Reply:
Updated